core/primitives/src/congestion_info.rs

```diff
@@ -57,18 +136,374 @@ impl Default for CongestionInfo {
 
 impl CongestionInfo {
     /// How much gas another shard can send to us in the next block.
-    pub fn outgoing_limit(&self, _sender_shard: ShardId) -> Gas {
-        todo!()
+    pub fn outgoing_limit(&self, sender_shard: ShardId) -> Gas {
+        match self {
+            CongestionInfo::V1(inner) => inner.outgoing_limit(sender_shard),
```

Comment:
This doesn't account for the missing chunks and the full info required for that is only available in the chunk header. I'm totally fine leaving that as is for now but eventually we'll need to combine both pieces.

Reply:
Yes. I think we discussed previously that we want to try and hide this detail from the runtime.

I though we can make necessary adjustments outside the runtime, when reading the congestion information from the header. Then we can set it to max congestion if we missed >3 chunks. That way, the code in this PR shouldn't need changes to provide the desired behaviour, I think.